src/test/java/ai/rev/speechtotext/unit/RevAiCustomVocabularyTest.java

```diff
@@ -0,0 +1,238 @@
+import ai.rev.speechtotext.CustomVocabulariesClient;
+import ai.rev.speechtotext.MockInterceptor;
+import ai.rev.speechtotext.models.CustomVocabulary;
+import ai.rev.speechtotext.models.CustomVocabularyInformation;
+import ai.rev.speechtotext.models.asynchronous.RevAiJobStatus;
+import com.google.gson.Gson;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class RevAiCustomVocabularyTest {
+
+  private OkHttpClient mockOkHttpClient;
+  private MockInterceptor mockInterceptor;
+  private CustomVocabulariesClient sut;
+
+  private final MediaType MEDIA_TYPE = MediaType.get("application/json; charset=utf-8");
+  private final String VOCABULARY_URL = "https://api.rev.ai/revspeech/v1/vocabularies";
+  private final String CALLBACK_URL = "https://test.callback.url";
+  private final String DATE = new Date().toString();
+  private final String ID = "TEST";
+  private Gson gson;
+
+  @Rule public TestName testName = new TestName();
+
+  @Before
+  public void setup() {
+    gson = new Gson();
+    mockInterceptor = new MockInterceptor(MEDIA_TYPE, 200);
+    sut = new CustomVocabulariesClient("validToken");
+    mockOkHttpClient = new OkHttpClient.Builder().addInterceptor(mockInterceptor).build();
+    Retrofit mockRetrofit =
+        new Retrofit.Builder()
+            .baseUrl("https://api.rev.ai/revspeech/v1/")
+            .addConverterFactory(GsonConverterFactory.create())
+            .client(mockOkHttpClient)
+            .build();
+    sut.apiInterface = mockRetrofit.create(ApiInterface.class);
+  }
+
+  @Test
+  public void
+      submitCustomVocabulary_WhenVocabularyCallbackMetadataAreSpecified_ReturnsACustomVocabularyInformation()
+          throws IOException {
+    CustomVocabularyInformation mockCustomVocabularyInfo = new CustomVocabularyInformation();
+    mockCustomVocabularyInfo.setCallbackUrl(CALLBACK_URL);
+    mockCustomVocabularyInfo.setCreatedOn(DATE);
+    mockCustomVocabularyInfo.setId(ID);
+    mockCustomVocabularyInfo.setStatus(RevAiJobStatus.IN_PROGRESS);
+    mockCustomVocabularyInfo.setMetadata(testName.getMethodName());
+
+    mockInterceptor.setSampleResponse(gson.toJson(mockCustomVocabularyInfo));
+
+    List<String> phrases = Arrays.asList("test", "custom", "vocabulary");
+    CustomVocabulary customVocabulary = new CustomVocabulary(phrases);
+
+    CustomVocabularyInformation customVocabularyInformation =
+        sut.submitCustomVocabularies(
+            testName.getMethodName(), Collections.singletonList(customVocabulary), CALLBACK_URL);
+
+    assertRequest(VOCABULARY_URL, "POST");
```

Comment:
I think you're missing an assertion on the request body. It seems we're only checking for url and apiMethod